tests/integration/jsdom/header.test.mjs

```diff
@@ -59,212 +75,209 @@ describe('Header class', () => {
         </nav>
       </div>
     `
-  })
 
-  it('Should create navigation elements in the DOM', async () => {
-    // Call the Header initialization function
-    await Header()
+    const container = document.querySelector('.nhsuk-navigation-container')
 
-    // Ensure the navigation elements are created in the DOM
-    expect(document.querySelector('.nhsuk-navigation')).not.toBeNull()
+    navigation = getByRole(container, 'navigation')
+    menuButton = getByRole(container, 'button', { name: 'Browse More' })
+
+    listWidth = 800
+    itemWidth = 100
+
+    jest
+      .spyOn(HTMLElement.prototype, 'offsetWidth', 'get')
+      .mockImplementation(function () {
+        return this instanceof HTMLUListElement
```

Comment:
[minor] Is there a way we could make this mock more targeted, like only applying it applying to a particular class or test id?